benchmark/protocol/src/main/scala/Protocol.scala

```diff
@@ -59,7 +59,7 @@ object protocol {
       .codec[MediumW]
       .codec[SmallW]
       .codec[BigW]
-      .specify1[LargeW, LargeW]("identityLarge")
-      .specify1[MediumW, MediumW]("identityMedium")
-      .specify1[BigW, BigW]("identityBig")
+      .specify1("identityLarge", Field.strict[LargeW]("in"), Type[LargeW])
```

Comment:
@jedesah A macro that does this is an interesting idea. However, is it not something that can be added later if the underlying model is sound and it is mostly about providing a nice api?